python/ray/autoscaler/sdk.py

```diff
@@ -178,6 +178,8 @@ def request_resources(num_cpus=None, bundles=None):
 
 
 def _as_config_file(cluster_config: Union[dict, str]):
+    # Ensure the temporary object outlives this function
+    global tmp
```

Comment:
Hmm, this should be done by making `_as_config_file` a `with` statement that is active while the file is needed.

Reply:
Done!